kore/src/Kore/Step/Remainder.hs

```diff
@@ -84,31 +52,34 @@ remainder
     => MultiOr (PredicateSubstitution Object (Target variable))
     -> Predicate Object variable
 remainder results =
-    unwrapRemainderVariables
-    $ mkMultiAndPredicate
-    $ mkNotMultiOr conditions
+    mkMultiAndPredicate $ mkNotExists conditions
```

Comment:
> I guess then that it's valid to drop all substitutions for target variables because we know that, if we have such a substitution, we already applied it so that variable does not occur anywhere else, right?

Yes, the target variables don't occur in the original configuration.

Now that you mention it, though: it would be safer to construct the remainder predicate *without* quantifiers here, and add the quantifiers at the top using `Exists.makeEvaluate`. It's a little more work to do that, but I think I like it better because it makes the `\exists`-simplifier into the single source of truth about what is a simplified `\exists` pattern.

Reply:
> using `Exists.makeEvaluate`

Actually, I will also need `Not.makeEvaluate` to do this right, so I'm going to wait until #594 is merged.